includes/admin/feedzy-rss-feeds-log.php

```diff
@@ -0,0 +1,858 @@
+	 * @param int                  $level The log level.
+	 * @param string               $message The log message.
+	 * @param array<string, mixed> $context The log context.
+	 * @return void
+	 */
+	public static function log( $level, $message, array $context = array() ) {
+		$instance = self::get_instance();
+		$instance->write_log( $level, $message, $context );
+	}
+
+	/**
+	 * Write log to file.
+	 *
+	 * @since 5.1.0
+	 * @param int                  $level The log level.
+	 * @param string               $message The log message.
+	 * @param array<string, mixed> $context The log context.
+	 * @return void
+	 */
+	private function write_log( $level, $message, array $context = array() ) {
+		if ( self::ERROR === $level ) {
+			$this->increment_log_stat( 'error_count' );
+			$this->try_accumulate_error_message( $message );
+		}
+		
+		if ( ! $this->filesystem ) {
+			return;
+		}
+
+		if ( $this->level_threshold > $level ) {
+			return;
+		}
+
+		$record = array(
+			'timestamp' => gmdate( 'c' ),
+			'level'     => isset( self::$levels[ $level ] ) ? self::$levels[ $level ] : 'UNKNOWN',
+			'message'   => $message,
+			'context'   => array_merge( $this->context, $context ),
+		);
+
+		if ( wp_doing_ajax() ) {
+			$record['doing_ajax'] = true;
+		}
+
+		if ( wp_doing_cron() ) {
+			$record['doing_cron'] = true;
+		}
+		
+		// Write log entry
+		$formatted = wp_json_encode( $record, JSON_UNESCAPED_SLASHES | JSON_UNESCAPED_UNICODE ) . PHP_EOL;
+
+		// Ensure the directory exists before writing.
+		$log_dir = dirname( $this->filepath );
+		if ( ! $this->filesystem->is_dir( $log_dir ) ) {
+			$this->setup_log_directory();
+		}
+
+		error_log( $formatted, 3, $this->filepath );
+	}
+
+	/**
+	 * Log a debug message.
+	 *
+	 * @since 5.1.0
+	 * @param string               $message The log message.
+	 * @param array<string, mixed> $context The log context.
+	 * @return void
+	 */
+	public static function debug( $message, array $context = array() ) {
+		self::log( self::DEBUG, $message, $context );
+	}
+
+	/**
+	 * Log an info message.
+	 *
+	 * @since 5.1.0
+	 * @param string               $message The log message.
+	 * @param array<string, mixed> $context The log context.
+	 * @return void
+	 */
+	public static function info( $message, array $context = array() ) {
+		self::log( self::INFO, $message, $context );
+	}
+
+	/**
+	 * Log a warning message.
+	 *
+	 * @since 5.1.0
+	 * @param string               $message The log message.
+	 * @param array<string, mixed> $context The log context.
+	 * @return void
+	 */
+	public static function warning( $message, array $context = array() ) {
+		self::log( self::WARNING, $message, $context );
+	}
+
+	/**
+	 * Log an error message.
+	 *
+	 * @since 5.1.0
+	 * @param string               $message The log message.
+	 * @param array<string, mixed> $context The log context.
+	 * @return void
+	 */
+	public static function error( $message, array $context = array() ) {
+		self::log( self::ERROR, $message, $context );
+	}
```

Comment:
It will be great to have a trace of the error by including the first parameter to be `__FUNCTION__` specially if it's made for debugging so one can easily debug which method threw the error.